demo/src/test/java/com/example/demo/service/ShopItemServiceTest.java

```diff
@@ -0,0 +1,105 @@
+    void shouldFindByPrice() {
+        //given
+        final var name1 = "banana";
+        final var priceOfItem = 10;
+        final Date date = null;
+
+        //when
+        when(webshopRepository.findByPrice(priceOfItem)).thenReturn(
+                List.of(ShopItem.builder()
+                        .id(1)
+                        .name(name1)
+                        .price(priceOfItem)
+                        .quantity(1)
+                        .date(null)
+                        .build()));
+        final var shopItems = webshopRepository.findByPrice(priceOfItem);
+
+        //then
+        Assertions.assertEquals(1, shopItems.size());
+        Assertions.assertEquals(name1, shopItems.get(0).getName());
+        Assertions.assertEquals(priceOfItem, shopItems.get(0).getPrice());
+    }
+
+    @Test
+    void shouldFindByPriceAndQuantity() {
+        //given
+        final var quantityOfItem = 5;
+        final var priceOfItem = 10;
+        final Date date = null;
+
+        //when
+        when(webshopRepository.findByPriceAndQuantity(priceOfItem, quantityOfItem)).thenReturn(
+                List.of(ShopItem.builder()
+                        .id(1)
+                        .name("banana")
+                        .price(priceOfItem)
+                        .quantity(quantityOfItem)
+                        .date(null)
+                        .build()));
+        final var shopItems = webshopRepository.findByPriceAndQuantity(priceOfItem, quantityOfItem);
+
+        //then
+        Assertions.assertEquals(1, shopItems.size());
+        Assertions.assertEquals(quantityOfItem, shopItems.get(0).getQuantity());
+        Assertions.assertEquals(priceOfItem, shopItems.get(0).getPrice());
+    }
+
+    @Test
+    void shouldFindByNameOrName() {
+        //given
+        final var nameOfItemA = "banana";
+        final var nameOfItemB = "cherry";
+        final Date date = null;
+
+        //when
+        when(webshopRepository.findByNameOrName(nameOfItemB, nameOfItemA)).thenReturn(
+                List.of(ShopItem.builder()
+                                .id(1)
+                                .name(nameOfItemA)
+                                .price(1)
+                                .quantity(1)
+                                .date(null)
+                                .build(),
+                        ShopItem.builder()
+                                .id(2)
+                                .name(nameOfItemB)
+                                .price(2)
+                                .quantity(2)
+                                .date(null)
+                                .build())
+        );
+        final var shopItems = webshopRepository.findByNameOrName(nameOfItemB, nameOfItemA);
+
+        //then
+        Assertions.assertEquals(2, shopItems.size());
+        Assertions.assertEquals(nameOfItemA, shopItems.get(0).getName());
+        Assertions.assertEquals(nameOfItemB, shopItems.get(1).getName());
+    }
+
```

Comment:
your tests still do not test what they should - please review carefully and adjust your tests

Reply:
It is partially solved, it will be continued.